get_e2e_test.go

```diff
@@ -670,6 +689,16 @@ func TestGet(t *testing.T) {
 					{
 						name: "Get tricky case with replace (thanos)",
 						do: func(t *testing.T) {
+							// In Go 1.16, 'go get' does not give hints about the versions it finds:
+							//     $ go get -modfile=thanos.mod -d github.com/thanos-io/thanos/cmd/thanos@f85e4003ba51f0592e42c48fdfdf0b800a23ba74
+							//             github.com/thanos-io/thanos/cmd/thanos imports
+							//             github.com/cortexproject/cortex/pkg/frontend imports
+							//             github.com/cortexproject/cortex/pkg/util imports
+							//             google.golang.org/grpc/naming: cannot find module providing package google.golang.org/grpc/naming
+							if !goVersion.LessThan(version.Go116) {
+								t.Skip("skipping pre-modules test in Go >= 1.16; the output of 'go get' does not give hints on the resolved versions")
```

Comment:
I don't think that is a solution, ideally we fix this too, right? ( :

Reply:
Sure, if the version info can be found yes, let's get this fixed as well

Reply:
Will look around